packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -139,28 +139,32 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
       )
 
       if (!this.disableQueueBatchAppend) {
-        const nonce = await this.signer.getTransactionCount()
+        // Generate the transaction we will repeatedly submit
+        const tx = await this.chainContract.populateTransaction.appendQueueBatch(
+          ethers.constants.MaxUint256 // Completely empty the queue by appending (up to) an enormous number of queue elements.
+        )
         const contractFunction = async (
           gasPrice
         ): Promise<TransactionReceipt> => {
           this.logger.info('Submitting appendQueueBatch transaction', {
             gasPrice,
-            nonce,
             contractAddr: this.chainContract.address,
           })
-          const tx = await this.chainContract.appendQueueBatch(99999999, {
-            nonce,
-            gasPrice,
-          })
+          const txResponse = await this.chainContract.appendQueueBatch(
+            ethers.constants.MaxUint256, // Completely empty the queue by appending (up to) an enormous number of queue elements.
+            {
+              gasPrice,
+            }
+          )
```

Comment:
you're directly calling the contract transaction here isntead of calling `sendTransaction` - is that intentional? given how the others are now all `sendTransaction`, and i'm wondering if the nonce increment bug would happen here

Reply:
Refactored this and it should no longer call this function directly!